rim/views.py

```diff
@@ -12,14 +13,22 @@
 from rim.models import Equipment, Checkout, EquipmentType, Location, Client
 from rim.forms import EquipmentForm
 
-class HomeView(ListView):
+class PaginateView(object):
+    def get_paginate_by(self, queryset):
+        if 'paginate' in self.request.COOKIES:
+            return self.request.COOKIES['paginate']
```

Comment:
You should be doing some additional validation here. What if someone set their cookie to negative numbers or infinity or 1.666 or "asdf"?

Reply:
If it *is* an invalid cookie, [tell the client to delete it](https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpResponse.delete_cookie).